src/api/connection.js

```diff
@@ -27,7 +47,21 @@ Connection_.prototype.buildRequestOptions = function (query) {
 }
 
 Connection_.prototype.fetchQuery = function (query) {
-  return UrlFetchApp.fetch(this.url, this.buildRequestOptions(query))
+  const cache = CacheService.getScriptCache()
+  const reformedQuery = this.buildRequestOptions(query)
+  const key = this.hash(JSON.stringify(reformedQuery))
+  const cachedResponse = cache.get(key)
+  if (cachedResponse !== null) {
+    return JSON.parse(cachedResponse)
```

Comment:
I wonder what happens with the cache expiration after a get operaton. Maybe you can do 
```js
cache.put(key, cachedResponse, 21600)
```
before returning to extend the expiration time